lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -4009,8 +4024,14 @@ LogicalResult BitCastOp::verify() {
   auto resTypeBits = getBitWidth(getType());
   if (inTypeBits.has_value() && resTypeBits.has_value()) {
     // Bitwidths must match for valid bit
-    if (*inTypeBits == *resTypeBits)
+    if (*inTypeBits == *resTypeBits) {
+      // non-'const' cannot be casted to 'const'
+      if (containsConst(getType()) && !isConst(getOperand().getType()))
```

Comment:
I'm not sure how important this is (might be but might very much not be), doesn't seem like we use bitcast a lot?

Reply:
The idea is mainly to disallow non-const to const as a basic sanity-check without doing the full const-compatible check, since a const-cast would be more appropriate.

But on the other hand, bitcasts seem to be purely internal as opposed to an actual firrtl op and aren't introduced until the lower-types pass, which is after where the drop-const pass will be, so maybe I don't need to do anything for this op?